### ISSUE TITLE
Support LastFm scrobbling

### ISSUE THREAD

**Issue by himynameisjonas** [FrontierPsychiatrist/node-spotify#91]
I would be nice if the `sp_session_set_social_credentials` method was available from the node api so we can scrobble played tracks to LastFm.

**Comment by FrontierPsychiatrist** [comment]
Yup, good idea.

**Comment by FrontierPsychiatrist** [comment]
Added to development, will be released with 0.8.0.

Added the following:
```javascript
spotify.constants = {
   //...
   SOCIAL_PROVIDER_SPOTIFY: ...,
   SOCIAL_PROVIDER_FACEBOOK: ...,
   SOCIAL_PROVIDER_LASTFM: ...,

   SCROBBLING_STATE_USE_GLOBAL_SETTING: ...,
   SCROBBLING_STATE_LOCAL_ENABLED: ...,
   SCROBBLING_STATE_LOCAL_DISABLED: ...,
   SCROBBLING_STATE_GLOBAL_ENABLED: ...,
   SCROBBLING_STATE_GLOBAL_DISABLED: ...,
   //...
}

/**
 * Takes one of spotify.constants.SOCIAL_PROVIDER_*
 * Returns one of spotify.constants.SCROBBLING_STATE_*
*/
spotify.isScrobbling(provider);

/**
 * Currently only accepts spotify.constants.SOCIAL_PROVIDER_FACEBOOK (see libspotify)
 * Returns a boolean signifying if scrobbling is possible
*/
spotify.isScrobblingPossible(provider);

/**
 * Takes one of spotify.constants.SOCIAL_PROVIDER_* and spotify.constants.SCROBBLING_STATE_LOCAL*
 *   Only the local settings are allowed!
*/
spotify.setScrobbling(provider, state);

/**
 * Takes one of spotify.constants.SOCIAL_PROVIDER_*
 * Only needed for last.fm if I get the libspotify documentation correctly
*/
spotify.setSocialCredentials(provider, username, password);
```

As always, only use these after logging in.

I'm not completely sure about the difference between LOCAL and GLOBAL scrobbling settings, I guess there are settings somewhere in the user profile?
